src/os/bluestore/BitMapAllocator.cc

```diff
@@ -31,9 +31,6 @@ BitMapAllocator::BitMapAllocator(int64_t device_size)
   m_bit_alloc = new BitAllocator(device_size / block_size,
         zone_size_blks, CONCURRENT, true);
   assert(m_bit_alloc);
-  if (!m_bit_alloc) {
-    dout(10) << __func__ << "Unable to intialize Bit Allocator" << dendl;
```

Comment:
I would suggest to change order. Also assert can be disabled non debug build. So it is not truly unreachable.